apps/api/src/models/post.ts

```diff
@@ -0,0 +1,49 @@
+import mongoose, { Document, Schema } from 'mongoose';
+
+interface IPost extends Document {
+  title: string;
+  image: string;
+  description: string;
+  category: string;
+  comments: string[];
+}
+
+export const postSchema = new Schema<IPost>(
+  {
+    title: {
+      type: String,
+      required: [true, 'Title is required'],
+      maxlength: [256, 'title should be no greater than 256 characters']
+    },
+    image: {
+      type: String,
+      required: [true, 'image URL is missing']
+    },
+    description: {
+      type: String,
+      required: [true, 'Description is required']
+    },
+    category: {
+      type: String,
```

Comment:
```suggestion
      type: mongoose.Schema.Types.ObjectId,
```

Since this property depends on a different model we can treat it as an `ObjectId` that way MongoDB will make the connection automatically